contrib/guix/build.sh

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env bash
+export LC_ALL=C
+
+set -ex
+
+if [ -z "$HOST" ]
+then
+    echo '$HOST: Environment variable not set.' >&2
+    exit 1
+fi
+
+# Guix-specific environment variables
+export GUIX_LD_WRAPPER_DISABLE_RPATH=yes
+
+CROSS_GLIBC="$(grep -E "/\S+glibc-cross-$HOST-[^-]+\\\"" "$GUIX_ENVIRONMENT/manifest" | sed -E -e 's|^\s+"||g' -e 's|"\s*$||g')"
+CROSS_GLIBC_STATIC="$(grep -E "/\S+glibc-cross-$HOST-[^-]+-static" "$GUIX_ENVIRONMENT/manifest" | sed -E -e 's|^\s+"||g' -e 's|"\s*$||g')"
+CROSS_KERNEL="$(grep -E "/\S+linux-libre-headers-cross-$HOST" "$GUIX_ENVIRONMENT/manifest" | head -n1 | sed -E -e 's|^\s+"||g' -e 's|"\s*$||g')"
+CROSS_GCC="$(grep -E "/\S+gcc-cross-$HOST" "$GUIX_ENVIRONMENT/manifest" | sed -E -e 's|^\s+"||g' -e 's|"\s*$||g')"
+
+export CROSS_C_INCLUDE_PATH="${CROSS_GCC}/include:${CROSS_GLIBC}/include:${CROSS_KERNEL}/include"
+export CROSS_CPLUS_INCLUDE_PATH="${CROSS_GCC}/include/c++:${CROSS_GLIBC}/include:${CROSS_KERNEL}/include"
+export CROSS_LIBRARY_PATH="${CROSS_GLIBC}/lib:${CROSS_GLIBC_STATIC}/lib:${CROSS_GCC}/lib:${CROSS_GCC}/${HOST}/lib:${CROSS_KERNEL}/lib"
+
+# Make /usr/bin if it doesn't exist
+[ -e /usr/bin ] || mkdir -p /usr/bin
+
+# Symlink file to a conventional path
+[ -e /usr/bin/file ] || ln -s "$(command -v file)" /usr/bin/file
+[ -e /usr/bin/env ]  || ln -s "$(command -v env)"  /usr/bin/env
+
+# Emulate Gitian reference date time logic
+if [ -z "$REFERENCE_UNIX_TIMESTAMP" ]
+then
+    echo '$REFERENCE_UNIX_TIMESTAMP: Environment variable not set.' >&2
+    exit 1
+fi
+REFERENCE_DATE="$(date --date="@${REFERENCE_UNIX_TIMESTAMP}" +'%F')"
+export REFERENCE_DATE
+REFERENCE_TIME="$(date --date="@${REFERENCE_UNIX_TIMESTAMP}" +'%T')"
+export REFERENCE_TIME
+export REFERENCE_DATETIME="${REFERENCE_DATE} ${REFERENCE_TIME}"
+
+# Setup an output directory
+OUTDIR="${PWD}/output"
+mkdir -p "${OUTDIR}"
+
+# Environment variables for determism
+export QT_RCC_TEST=1
+export QT_RCC_SOURCE_DATE_OVERRIDE=1
+export TAR_OPTIONS="--owner=0 --group=0 --numeric-owner --mtime='@${REFERENCE_UNIX_TIMESTAMP}' --sort=name"
```

Comment:
In #16141, `TAR_OPTIONS` has been removed from the gitian descriptor. Do we want to do the same here?

Reply:
It seems that change breaks a few things... I've made a few cleanups to how we archive in this script that can be cleanly backported to Gitian, so I'll go take a look and fix things up.